### ISSUE TITLE
ximea's unsupported temperature selector enums give different error codes

### ISSUE THREAD

**Issue by carandraug** [python-microscope/vendor-issues#6]
With Ximea's camera, one can get temperature measurements from different sensors.  The sensor to use is set by calling set_temp_selector (via Python's API).  Not all cameras have the same set of temperature sensors and calling this with an unsupported sensor will raise an error.  However, the error returned is not always the same.  In some case it returns error code 12 ("Not supported") and other times it returns error code 100 ("Unknown parameter").

Here's the code to replicate the issue (with a XiC camera) on a live Python session:

```
import ximea.xiapi
import ximea.xidefs

handle = ximea.xiapi.Camera()
handle.open_device()

sensor_board = "XI_TEMP_SENSOR_BOARD"
interface_board = "XI_TEMP_INTERFACE_BOARD"
sensor_die = "XI_TEMP_IMAGE_SENSOR_DIE"

assert sensor_board in ximea.xidefs.XI_TEMP_SELECTOR
handle.set_temp_selector(sensor_board)  # this works

assert interface_board in ximea.xidefs.XI_TEMP_SELECTOR
handle.set_temp_selector(interface_board)  # raises error 12

assert sensor_die in ximea.xidefs.XI_TEMP_SELECTOR
handle.set_temp_selector(sensor_die)  # raises error 100
```

This has been reported to Ximea's (their internal and private ticket system number 56130) and they have confirmed it. They plan on fixing this on a later version:

> thank you for notifying us about this ambiguity. Each enumerator  item which returns error is not supported by this camera model, but we will unify the error prints.

**Comment by carandraug** [comment]
This is fixed since version 4.21.19. Their reply:

> it has been fixed, now all unsupported temp. selector items return Error 100 - unknown parameter.

Why they didn't choose the "Not supported" error for the case of "unsupported" is a mystery. Their reply when asked about it:

> I admit that this particular error description is not very intuitive and we are planning to review all error messages in the future.

**Comment by carandraug** [comment]
It appears that ximea has indeed reviewed those errors and it's now returning error 26 (Not implemented). See report of this new behaviour in python-microscope/microscope#239

Current state is that depending on xiapi version and camera model, a non supported temperature sensor might return 12 (not supported), 26 (not implemented), or 100 (unknown parameter).
